internal/services/domainservices/active_directory_domain_service_trust_resource.go

```diff
@@ -0,0 +1,350 @@
+package domainservices
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/services/domainservices/mgmt/2020-01-01/aad"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/locks"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/domainservices/parse"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/domainservices/validate"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+)
+
+type DomainServiceTrustResource struct{}
+
+var _ sdk.ResourceWithUpdate = DomainServiceTrustResource{}
+
+type DomainServiceTrustModel struct {
+	Name                    string   `tfschema:"name"`
+	DomainServiceResourceId string   `tfschema:"domain_service_resource_id"`
+	TrustedDomainFqdn       string   `tfschema:"trusted_domain_fqdn"`
+	TrustedDomainDnsIPs     []string `tfschema:"trusted_domain_dns_ips"`
+	Password                string   `tfschema:"password"`
+}
+
+func (r DomainServiceTrustResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+		"domain_service_resource_id": {
```

Comment:
We can shorten this to `domain_service_id`

```suggestion
		"domain_service_id": {
```